### PR TITLE
Fix for #31

### LINKED ISSUE #31: Filename regex for prefetching does not match
There are still some packages whose filenames don’t match the regex and are ignored (see https://github.com/anatol/pacoloco/pull/28#pullrequestreview-741066693). It seems that the packages are from a third party repository. See this log excerpt:

```
repo_db_mirror.go:184: Adding entries to db...
repo_db_mirror.go:191: error: filename android-sdk-26.1.1-1.1-x86_64.pkg.tar.xz does not match regex, matches length is 0
repo_db_mirror.go:191: error: filename grub2-theme-arch-leap-1-3.1-any.pkg.tar.xz does not match regex, matches length is 0
repo_db_mirror.go:191: error: filename hpoj-0.91-21.1-x86_64.pkg.tar.xz does not match regex, matches length is 0
repo_db_mirror.go:191: error: filename libnatspec-0.3.0-2.1-x86_64.pkg.tar.xz does not match regex, matches length is 0
repo_db_mirror.go:191: error: filename qt5-base-24h-5.15.2+kde+r254-1.7-x86_64.pkg.tar.zst does not match regex, matches length is 0
repo_db_mirror.go:191: error: filename qt5-base-24hms-5.15.2+kde+r254-1.7-x86_64.pkg.tar.zst does not match regex, matches length is 0
repo_db_mirror.go:191: error: filename rxvt-unicode-better-wheel-scrolling-9.22-5.1-x86_64.pkg.tar.zst does not match regex, matches length is 0
repo_db_mirror.go:191: error: filename vim-pathogen-2.4-1.1-any.pkg.tar.xz does not match regex, matches length is 0
repo_db_mirror.go:220: Added entries to db.
```

### PR DESCRIPTION
**Pull request by Focshole**
I had never seen dots in the "subversion" field before. I had included them as allowed chars for that field.

Fixes #31

### CODE REVIEW

**Review by Focshole — COMMENTED** [review]
This should fix issue https://github.com/anatol/pacoloco/issues/31 .

**Review by anatol — COMMENTED** [review]
Would it be possible to add a unit test for this issue?

File: pacoloco.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func init() {
 
 
 	*/
-	filenameRegex, err = regexp.Compile("^([a-z0-9._+-]+)-([a-zA-Z0-9:._+]+-[0-9]+)-([a-zA-Z0-9:._+]+)(([.]pkg[.]tar(([.]gz)|([.]bz2)|([.]xz)|([.]zst)|([.]lzo)|([.]lrz)|([.]lz4)|([.]lz)|([.]Z))?)([.]sig)?)$")
+	filenameRegex, err = regexp.Compile("^([a-z0-9._+-]+)-([a-zA-Z0-9:._+]+-[0-9.]+)-([a-zA-Z0-9:._+]+)(([.]pkg[.]tar(([.]gz)|([.]bz2)|([.]xz)|([.]zst)|([.]lzo)|([.]lrz)|([.]lz4)|([.]lz)|([.]Z))?)([.]sig)?)$")
```
Comment (anatol) [inline pacoloco.go:73]: "dot" in a regex means "any symbol. Most likely you need to escape this and other "dot symbols", like `\.`.

**Review by anatol — COMMENTED** [review]
Could you please improve the commit description? Add a not about what is the issue and what was done to fix it. At the end add "Fixes #31" marker.

And if possible could you please add a unit test as well?

File: pacoloco.go (hunk 1 above)
Comment (Focshole) [reply pacoloco.go:73]: https://stackoverflow.com/questions/19976018/does-a-dot-have-to-be-escaped-in-a-character-class-square-brackets-of-a-regula
Not into square brackets, it is fine. I'll add a unit test which does test that case anyway.

**Review by Focshole — COMMENTED** [review]
It should be fine now

**Comment by anatol** [comment]
Thank you. This PR is merged as https://github.com/anatol/pacoloco/commit/01baa2214b646cdb1ec840cc4e8c9757d6d714a8

**Pull request closed without merging**